java-manta-client-unshaded/src/main/java/com/joyent/manta/client/MantaClient.java

```diff
@@ -164,6 +164,11 @@
             = (Collections.newSetFromMap(new ConcurrentWeakIdentityHashMap<>()));
 
     /**
+     * Signing object used for authentication and signed URL generation.
+     */
+    private final ThreadLocalSigner signer;
```

Comment:
As voiced in chat, I have reservations about an object having a strong reference to a `ThreadLocal`. In all probability it is ok - it rubs me the wrong way because the reference isn't needed and I can imagine in cases with OSGI or servlet containers the object graph may get rather circular and the ThreadLocal may never be properly cleaned up thereby resulting in a memory leak. Defensive (or cargo cult) as it may be, I do not like to have any strong references to a ThreadLocal so that the path to proper GC is as simple as possible.

[More on ThreadLocal](https://blog.codecentric.de/en/2008/09/a-threadlocal-memory-leak/) - but nothing about references to it.

Reply:
Would using a `WeakReference` here like `MantaConnectionFactory` was using mitigate this concern?